src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.RSA.Mobile.cs

```diff
@@ -0,0 +1,198 @@
+            SafeSecKeyRefHandle privateKey,
+            ref byte pbData,
+            int cbData,
+            out SafeCFDataHandle pDataOut,
+            out SafeCFErrorHandle pErrorOut);
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_RsaVerificationPrimitive(
+            SafeSecKeyRefHandle publicKey,
+            ref byte pbData,
+            int cbData,
+            out SafeCFDataHandle pDataOut,
+            out SafeCFErrorHandle pErrorOut);
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_RsaDecryptionPrimitive(
+            SafeSecKeyRefHandle privateKey,
+            ref byte pbData,
+            int cbData,
+            out SafeCFDataHandle pDataOut,
+            out SafeCFErrorHandle pErrorOut);
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_RsaEncryptionPrimitive(
+            SafeSecKeyRefHandle publicKey,
+            ref byte pbData,
+            int cbData,
+            out SafeCFDataHandle pDataOut,
+            out SafeCFErrorHandle pErrorOut);
+
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        internal static void RsaGenerateKey(
+            int keySizeInBits,
+            out SafeSecKeyRefHandle pPublicKey,
+            out SafeSecKeyRefHandle pPrivateKey)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        internal static byte[] RsaEncrypt(
+            SafeSecKeyRefHandle publicKey,
+            byte[] data,
+            RSAEncryptionPadding padding)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        internal static bool TryRsaEncrypt(
+            SafeSecKeyRefHandle publicKey,
+            ReadOnlySpan<byte> source,
+            Span<byte> destination,
+            RSAEncryptionPadding padding,
+            out int bytesWritten)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        internal static byte[] RsaDecrypt(
+            SafeSecKeyRefHandle privateKey,
+            byte[] data,
+            RSAEncryptionPadding padding)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        internal static bool TryRsaDecrypt(
+            SafeSecKeyRefHandle privateKey,
+            ReadOnlySpan<byte> source,
+            Span<byte> destination,
+            RSAEncryptionPadding padding,
+            out int bytesWritten)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        private static bool ProcessPrimitiveResponse(
```

Comment:
This is all duplicated code from RSA.cs, right? Can we reduce that?

Reply:
Yeah, it should be possible to extract it to a shared part, thanks

Reply:
Addressed, thanks